android/autodispose-android/src/main/java/com/uber/autodispose/android/ViewScopeProvider.java

```diff
@@ -54,7 +55,7 @@
    * @param view the view to scope for
    * @return a {@link LifecycleScopeProvider} against this view.
    */
-  public static LifecycleScopeProvider from(View view) {
+  public static LifecycleScopeProvider<ViewLifecycleEvent> from(@Nullable View view) {
```

Comment:
This should not be `@Nullable`. Also, let's make the return type `<?>` to keep the `ViewLifecycleEvent` an implementation detail

Reply:
So what triggered this was the explicit check for null -- can that be removed? I can just leave it alone with the warning too.

Reply:
We have jsr305 package annotations to signal to static analysis that parameters are nonnull by default

We have the nullcheck as a fallback because we don't trust external users' static analysis and fail eagerly :). Just leave the warning